src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -376,19 +356,34 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
       (this.getNameRequestLegalName !== this.getEntitySnapshot?.businessInfo?.legalName)
   }
 
-  /** Check if current entity selection is a Benefit Company */
-  get isBenefit (): boolean {
-    return (this.selectedEntityType === CorpTypeCd.BENEFIT_COMPANY)
-  }
-
   /** Type change helper information */
   get typeChangeInfo (): string {
     return this.getResource.changeData?.typeChangeInfo
   }
 
+  /** Entity type options based on the company type */
+  get entityTypeOptions (): Array<{ value: string, SHORT_DESC: string, text: string }> {
+    if (this.isBcCompany) {
```

Comment:
This is already done in the alteration view, to choose the correct resource. If you moved the business types to the resources, you wouldn't need to do this.

Reply:
EG. `this.getResource.entityTypeOptions`